site/plugins/site/extensions/pageMethods.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+return [
+	/**
+	 * Checks whether the page is the current page in a menu context,
+	 * considering redirects to other pages
+	 */
+	'isActiveInMenu' => function (bool $hasSubmenu = false) {
```

Comment:
Wondering if `->isOpen()` wouldn't work just as well? https://getkirby.com/docs/reference/objects/cms/page/is-open